src/app/components/movies/movies.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit, Input, Output, EventEmitter, HostListener } from '@angular/core';
+import { Movie } from '../../models/movie.interface';
+import { Store, select } from '@ngrx/store';
+import { AppState } from '../../store/state/app.state';
+import { LoadPage } from '../../store/actions/movies-page.actions';
+import { selectCurrentPage, selectMoviesList } from '../../store/selectors/movies-page.selectors';
+import { SearchService } from '../../services/search.service';
+
+@Component({
+  selector: 'app-movies',
+  templateUrl: './movies.component.html',
+  styleUrls: ['./movies.component.sass']
+})
+export class MoviesComponent implements OnInit {
+  private nextPage: number;
+  private searchIsOff = true;
```

Comment:
what is searchIsOff

Reply:
it's switcher between filter search on the page and loading next chunk of movies after scroll. when search input isn't empty loading next movies is unavailabale